### ISSUE TITLE
Improve error for malformed signature

### ISSUE THREAD

**Issue by tbfleming** [EOSIO/eos#4214]
If a signature on a `push_transaction` is malformed (e.g. too many bits), then the error is confusingly similar to transaction serialization problems.

e.g. malformed signature: `SIG_R1_AN1rKqDi1AKzh6z4xrPfERmWvuEmnoY4wX1BWGMrGPAmAeaRMxJPQb3wfifxAAzREUXWDaZ9dfrcL9sQog83rDhTxvVYzhjdG`

**Comment by andriantolie** [comment]
When using the above signature in the current latest develop branch, I've got 
```
2018-08-09T09:14:31.759 thread-0   main.cpp:2890                 main                 ] Failed with error: Assert Exception (10)
checksum == wrapped.check:
```
instead of transaction serialization error. Does it mean this issue is already fixed by 1.1.4?

**Comment by tbfleming** [comment]
Look at the JSON response; that's what dapp devs tend to see.

**Comment by andriantolie** [comment]
Ah, I see. The JSON response returns
```
{
    "code": 500,
    "message": "Internal Service Error",
    "error": {
        "code": 3010010,
        "name": "packed_transaction_type_exception",
        "what": "Invalid packed transaction",
        "details": [
            {
                "message": "Invalid packed transaction",
                "file": "chain_plugin.cpp",
                "line_number": 1339,
                "method": "push_transaction"
            },
            {
                "message": "checksum == wrapped.check: ",
                "file": "common.hpp",
                "line_number": 48,
                "method": "apply"
            },
            {
                "message": "Failed to deserialize variant",
                "file": "abi_serializer.hpp",
                "line_number": 601,
                "method": "from_variant"
            }
        ]
    }
}
```
which I guess, we prefer it to show invalid signature instead of invalid packed transaction?

**Comment by arhag** [comment]
Resolved by #5305.
